src/LinearSolvers/preconditioners.jl

```diff
@@ -42,22 +42,22 @@ struct BlockJacobiPreconditioner{AT,GAT,VI,GVI,MT,GMT,MI,GMI,SMT} <: AbstractPre
     P::SMT
     id::Union{GMT,MT}
     function BlockJacobiPreconditioner(J, npart, device=CPU()) where {}
-        if device == CPU()
+        if isa(device, CPU)
             AT  = Array{Float64,3}
             GAT = Nothing
             VI  = Vector{Int64}
             GVI = Nothing
-            MT = Matrix{Float64}
+            MT  = Matrix{Float64}
             GMT = Nothing
             MI  = Matrix{Int64}
-            GMI  = Nothing
+            GMI = Nothing
             SMT = SparseMatrixCSC{Float64,Int64}
-        elseif device == CUDADevice()
+        elseif isa(device, GPU)
```

Comment:
Do we want to use `CuArray` here?

Reply:
Yes. It's used throughout the code. This is the place where `Preferences.jl` would come in, but I don't want to add that now. I want to add that when at least one kernel has a test on AMD GPUs.